index.js

```diff
@@ -36,11 +37,26 @@ var reservedRe = /^\.+$/;
 var windowsReservedRe = /^(con|prn|aux|nul|com[0-9]|lpt[0-9])(\..*)?$/i;
 var windowsTrailingRe = /[\. ]+$/;
 
-function sanitize(input, replacement) {
+function sanitize(input, replacement, invalids) {
   if (typeof input !== 'string') {
     throw new Error('Input must be string');
   }
-  var sanitized = input
+  if (typeof replacement === 'string' && invalids.indexOf(replacement) !== -1) {
+    throw new Error(`The replacement string can't be part of options.additionalInvalids or contain substrings which are part of options.additionalInvalids!`);
+  }
+  var sanitized = input;
+  while (invalids.some(invalid => sanitized.includes(invalid))) {
+    let counter = 0;
+    for (const invalid of invalids) {
+      sanitized = sanitized.split(invalid).join(replacement);
+    }
+    if (counter > 1000) {
+      throw new Error(`Illeagal replacement function. Make sure that replacements generated by your function do not contain any of the strings specified in options.additionalInvalids!`);
```

Comment:
When I saw the `while` loop, I was a little worried about the potential of accidental infinite loops. Nice job thinking that through and putting this guard in place!

Reply:
I'm so happy I made proper tests :D